ghost/core/test/e2e-api/admin/users.test.js

```diff
@@ -1,492 +1,600 @@
-        const res = await request
-            .get(localUtils.API.getApiQuery(`users/${userToEditId}/?include=roles`))
-            .set('Origin', config.get('url'))
-            .expect(200);
+        const userToEditId = fixtureManager.get('users', 1).id;
+        
+        // First, fetch the user with roles
+        const res = await agent.get(`users/${userToEditId}/?include=roles`)
+            .expectStatus(200)
+            .matchHeaderSnapshot({
+                'content-version': anyContentVersion,
+                etag: anyEtag
+            })
+            .matchBodySnapshot({
+                users: [{
+                    ...userMatcher,
+                    roles: anyArray,
+                    name: fixtureManager.get('users', 1).name
+                }]
+            });
 
         const jsonResponse = res.body;
-        jsonResponse.users[0].name.should.equal('Ghost');
-
-        should.exist(jsonResponse.users[0].roles);
-        jsonResponse.users[0].roles.should.have.length(1);
-        jsonResponse.users[0].roles[0].name.should.equal('Contributor');
+        const expectedRoleName = jsonResponse.users[0].roles[0].name;
 
+        // Modify the user data
         jsonResponse.users[0].name = 'Changed Name';
 
-        const editResponse = await request
-            .put(localUtils.API.getApiQuery(`users/${userToEditId}/?include=roles`))
-            .set('Origin', config.get('url'))
-            .send({
+        // Send the update
+        await agent.put(`users/${userToEditId}/?include=roles`)
+            .body({
                 users: jsonResponse.users
             })
-            .expect('Content-Type', /json/)
-            .expect('Cache-Control', testUtils.cacheRules.private)
-            .expect(200);
-
-        const editJSONResponse = editResponse.body;
-        editJSONResponse.users[0].name.should.equal('Changed Name');
-
-        should.exist(editJSONResponse.users[0].roles);
-        editJSONResponse.users[0].roles.should.have.length(1);
-        editJSONResponse.users[0].roles[0].name.should.equal('Contributor');
+            .expectStatus(200)
+            .matchHeaderSnapshot({
+                'content-version': anyContentVersion,
+                etag: anyEtag
+            })
+            .expect(cacheInvalidateHeaderSetToWildcard())
+            .matchBodySnapshot({
+                users: [{
+                    ...userMatcher,
+                    roles: anyArray,
+                    name: 'Changed Name'
+                }]
+            })
+            .expect(({body}) => {
+                assert.equal(body.users[0].roles[0].name, expectedRoleName);
+            });
     });
 
     it('Can edit user with empty roles data and does not change the role', async function () {
-        const res = await request.put(localUtils.API.getApiQuery('users/me?include=roles'))
-            .set('Origin', config.get('url'))
-            .send({
+        await agent.put('users/me?include=roles')
```

Comment:
This is missing a trailing slash so we'll have a hidden redirect here. I guess the agent just follows it considering tests are passing but adding the trailing slash means we aren't introducing subtle behaviour that's not explicitly tested/called out.